chatgpt-retrieval-plugin/run.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+# chat-gpt-retrieval-plugin is a poetry run project
+
+set -x
+
+. $workdir/src/.evergreen/utils.sh
+
+PYTHON_BINARY=$(find_python3)
+$PYTHON_BINARY -c "import sys; print(f'Python version found: {sys.version_info}')"
+
+# Install Poetry into base python
+$PYTHON_BINARY -m pip install -U pip poetry
+# Create a package specific poetry environment
+$PYTHON_BINARY -m poetry env use $PYTHON_BINARY
+# Activate the poetry env, which itself does not include poetry
+source $($PYTHON_BINARY -m poetry env info --path)/bin/activate
+# Recreate the poetry lock file
+#$PYTHON_BINARY -m poetry lock
+# Install from pyproject.toml into package specific environment
+$PYTHON_BINARY -m poetry install --with dev
+
+# Run tests. Sensitive variables in Evergreen come from Evergeen project: ai-ml-pipeline-testing/
+OPENAI_API_KEY=$openai_api_key \
+DATASTORE="mongodb" \
+BEARER_TOKEN="staylowandkeepmoving" \
+MONGODB_URI=$chatgpt_retrieval_plugin_mongodb_uri \
+MONGODB_DATABASE="chatgpt_retrieval_plugin_test_db" \
```

Comment:
OK cool. How do ever green task "expansions" work? What is their scope?

Reply:
They work like variables that go top-down. Where the expansion is first defined, it can then be used at inner call-sites. Here's a link to the [expansions](https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Project-Configuration-Files/#expansions) explanation.

I set that any expansion provided in the `config.yml` will also be placed as an environment variable. So when you defined `DATABASE: chatgpt_retrieval_plugin_test_db` in the `config.yml` file, it automatically made it an environment variable that can be used in any scripts run.